packages/application/src/status.ts

```diff
@@ -71,15 +72,28 @@ export class LabStatus implements ILabStatus {
    *
    * @returns A disposable used to clear the busy state for the caller.
    */
-  setBusy(): IDisposable {
+  setBusy(translator?: ITranslator): IDisposable {
+    translator = translator || nullTranslator;
+    const trans = translator.load('jupyterlab');
     const oldBusy = this.isBusy;
+    const ariaLiveRegion = document.getElementById('commands-aria-live');
+    let timeout: any;
     this._busyCount++;
     if (this.isBusy !== oldBusy) {
       this._busySignal.emit(this.isBusy);
+      timeout = setTimeout(() => {
+        ariaLiveRegion!.append(
+          trans.__('A notebook cell is processing. Kernel status is busy.')
+        );
```

Comment:
Do we need to clear the live region to avoid memory leak?

Reply:
yes. an activity log or aria log will need to be cleared. over the course of an interactive computing session the activity will fill up, especially after hours of work. memory leaks are inevitable in these cases.

assertive aria live notifications will be announced immediately and may be cleared sooner while polite aria live messages linger until current speaking buffer clears. this AT buffer could be reading an entire page or post meaning that polite aria messages may take many minutes to be announced. 
unfortunately, there is NO way to know when to clear the aria live buffer because the browser can't know when messages have reached assistive technology. for polite aria messages the buffer might need to persist for minutes maybe up tens of minutes; it is possible an aria live message will become stale if enough time and activity have elapsed. we can't make too many assumptions about time because for the varied ways disabled people experience crip time.